src/components/ShoppingList/ShoppingList.js

```diff
@@ -21,13 +28,24 @@ function ShoppingList({ listId }) {
         </div>
       )}
 
-      {!loading && listItems && (
+      {/* !loading is required to access listItems array */}
+      {/* listItems.empty is a Boolean for whether querySnapshot is empty */}
```

Comment:
So Nick and I ran into an issue where we tried to just check `!listItems.empty` and kept getting that the listItems array from Firestore was undefined unless we also checked it was `!loading`. That is what that comment is alluding to, but we can probably improve the language so it's clear.

Reply:
Ahh, I see. I think it's always a good idea to make sure something is an array before you use .empty, or you can get Type errors before it has been populated.

Reply:
Sorry to be super late in commenting on this @anderswift, the reason we don't check for the array is because the `listItems` is actually a query snapshot. The `.empty` is a boolean property on that snapshot provided by firebase. The array comes from the `.docs` method of the snapshot.

Reply:
@nick-zanetti Ah yes, you're totally right! I remember the Firestore empty property from previous sprints. I was at least 3/4 asleep mid-Hackathon when I wrote that, trying to quickly respond to all the github notifications I'd missed. 😂 In my head, I was thinking you were describing having a problem calling .length on an undefined array.